File: ffb-java/ffb-server/src/main/java/com/fumbbl/ffb/server/mechanic/WeatherModifierValues.java
package com.fumbbl.ffb.server.mechanic;

/**
 * Documented modifier values for weather effects on rolls.
 *
 * Sign convention (positive = harder for the rolling player, negative = easier):
 * - Dodge/catch/pass: modifier is ADDED to the roll target (min 2).
 *   +1 = harder; -1 = easier.
 * - This matches Java's DodgeModifierCollection/CatchModifierCollection/PassModifierCollection
 *   where tackle zones are +1 (harder) and skill bonuses like Accurate are -1 (easier).
 */
public final class WeatherModifierValues {

    /**
     * Pouring Rain: +1 to catch and pickup targets (harder to handle the ball).
     * Applied to catch and pickup rolls only; not to pass rolls.
     * Source: CatchModifierCollection "Pouring Rain" = +1.
     */
    public static final int POURING_RAIN_CATCH = 1;

    /**
     * Blizzard (BB2016): +0 to pass rolls.
     * Blizzard in BB2016 does not modify the pass roll directly (only movement and catching are affected).
     * Source: BB2016 PassModifierCollection "Blizzard" = 0.
     */
    public static final int BLIZZARD_PASS_BB2016 = 0;

    /**
     * Blizzard (BB2025): +1 to GFI minimum roll (needs 3+ instead of 2+).
     * Source: BB2025 GoForItModifierCollection "Blizzard" = +1.
     */
    public static final int BLIZZARD_GFI_BB2025 = 1;

    /**
     * Very Sunny: +1 to pass rolls (harder to pass in bright sunlight, glare).
     * Source: PassModifierCollection "Very Sunny" = +1.
     */
    public static final int VERY_SUNNY_PASS = 1;

    private WeatherModifierValues() {}
}

File: ffb-rust/crates/ffb-engine/src/mechanic/weather_modifier_values.rs
/// 1:1 translation of `com.fumbbl.ffb.server.mechanic.WeatherModifierValues`.
///
/// Documented modifier values for weather effects on rolls.
///
/// Sign convention (positive = harder for the rolling player, negative = easier):
/// - Dodge/catch/pass: modifier is ADDED to the roll target (min 2).
///   +1 = harder; -1 = easier.
pub struct WeatherModifierValues;

impl WeatherModifierValues {
    /// Pouring Rain: +1 to catch and pickup targets (harder to handle the ball).
    pub const POURING_RAIN_CATCH: i32 = 1;

    /// Blizzard (BB2016): +0 to pass rolls (no pass modifier in BB2016 Blizzard).
    pub const BLIZZARD_PASS_BB2016: i32 = 0;

    /// Blizzard (BB2025): +1 to GFI minimum roll (needs 3+ instead of 2+).
    pub const BLIZZARD_GFI_BB2025: i32 = 1;

    /// Very Sunny: +1 to pass rolls (harder to pass in bright sunlight).
    pub const VERY_SUNNY_PASS: i32 = 1;
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn pouring_rain_catch_is_1() {
        assert_eq!(WeatherModifierValues::POURING_RAIN_CATCH, 1);
    }

    #[test]
    fn blizzard_pass_bb2016_is_0() {
        assert_eq!(WeatherModifierValues::BLIZZARD_PASS_BB2016, 0);
    }

    #[test]
    fn blizzard_gfi_bb2025_is_1() {
        assert_eq!(WeatherModifierValues::BLIZZARD_GFI_BB2025, 1);
    }

    #[test]
    fn very_sunny_pass_is_1() {
        assert_eq!(WeatherModifierValues::VERY_SUNNY_PASS, 1);
    }
    #[test]
    fn blizzard_and_pouring_rain_are_different_effects() {
        assert_ne!(WeatherModifierValues::BLIZZARD_GFI_BB2025, WeatherModifierValues::BLIZZARD_PASS_BB2016);
    }
}
